### ISSUE TITLE
JsonIgnoreProperties is ignored

### ISSUE THREAD

**Issue by gjesse** [immutables/immutables#169]
I have a class similar to this, and I would expect it to work, but it does not.

```java
@JsonDeserialize(as = ImmutableTestIgnore.class)
@JsonSerialize(as = ImmutableTestIgnore.class)
@JsonIgnoreProperties(ignoreUnknown = true)
@Value.Immutable
public abstract class TestIgnore {
    public abstract String id();

    public static TestIgnore testSerial(String input) throws Exception{
        ObjectMapper m = new ObjectMapper();
        return m.readValue(input, TestIgnore.class);
    }

    public static void main(String[] args) throws Exception {

        String input = "{\"id\":\"some-id\"}";
        System.out.println(testSerial(input));

        input = "{\"id\":\"some-id\",\"additionalField\":22}";
        System.out.println(testSerial(input));
    }

}
```

With an error message like so:

```
Exception in thread "main" com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "additionalField" (class ...ImmutableTestIgnore$Json), not marked as ignorable (one known property: "id"])
```

Would it be possible for the Immutable generated classes to preserve the @JsonIgnoreProperties annotation?

**Comment by elucash** [comment]
We should support this, so it's definitely a bug. Thank you for reporting this!

**Comment by thelastnode** [comment]
:+1:

**Comment by elucash** [comment]
Fix Released in 2.1.0

**Comment by gjesse** [comment]
Thanks!
